pkg/agent/agent.go

```diff
@@ -161,12 +163,27 @@ func OnAgentResourceChange(agentResourceChangeHandler ConfigChangeHandler) {
 
 func startAPIServiceCache() {
 	// register the update cache job
-	id, err := jobs.RegisterIntervalJob(&discoveryCache{}, agent.cfg.GetPollInterval())
+	allDiscoveryCacheJob := newDiscoveryCache(true)
+	id, err := jobs.RegisterIntervalJobWithName(allDiscoveryCacheJob, time.Hour, "All APIs Cache")
 	if err != nil {
-		log.Errorf("could not start the API cache update job: %v", err.Error())
+		log.Errorf("could not start the All APIs cache update job: %v", err.Error())
 		return
 	}
-	log.Tracef("registered API cache update job: %s", id)
+	log.Tracef("registered API cache update all job: %s", id)
+
+	// Start the regular update after the first interval
+	go func() {
+		newDiscoveryCacheJob := newDiscoveryCache(false)
+		newDiscoveryCacheJob.lastInstanceTime = time.Now()
+		newDiscoveryCacheJob.lastServiceTime = time.Now()
+		time.Sleep(agent.cfg.GetPollInterval())
```

Comment:
Will having this configurable be an issue? Haven't checked, but do we validate this config for < 0 or non numeric

Reply:
Has to be a positive value, > 0

Reply:
Also the cfg library has it set as an int, so that validation is done

Reply:
Too lazy to check just wanted to make sure they can't do -1

Reply:
We, as discussed, should still add a floor, as 1ns is valid....